crates/components-runtime/src/config.rs

```diff
@@ -24,18 +24,31 @@ pub struct DataCollectionComponents {
 
 #[derive(Deserialize, Debug, Clone)]
 pub struct DataCollectionComponentConfig {
+    #[serde(default = "default_true")]
     pub anonymization: bool,
+    #[serde(default = "default_consent")]
     pub default_consent: String,
+    #[serde(default = "default_true")]
     pub track_event_enabled: bool,
+    #[serde(default = "default_true")]
     pub user_event_enabled: bool,
+    #[serde(default = "default_true")]
     pub page_event_enabled: bool,
 }
```

Comment:
Using `#[serde(default)]` would do the trick here:
```suggestion
#[serde(default)]
pub struct DataCollectionComponentConfig {
    pub anonymization: bool,
    pub default_consent: String,
    pub track_event_enabled: bool,
    pub user_event_enabled: bool,
    pub page_event_enabled: bool,
}
```